tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle@2023_01_11/block_invoke.pysnap

```diff
@@ -0,0 +1,43 @@
+---
+created: '2023-12-12T18:43:32.477116Z'
+creator: sentry
+source: tests/sentry/grouping/test_variants.py
+---
+app:
+  hash: null
+  component:
+    app (threads of system take precedence)
+      threads (ignored because hash matches system variant)
+        stacktrace*
+          frame* (marked in-app by stack trace rule (family:native package:**/containers/bundle/application/** +app))
+            function*
+              "__46+[FudgeGlobalHandler setupGlobalHandlersIfNeeded]_block_invoke_2"
+          frame* (marked in-app by stack trace rule (family:native package:**/containers/bundle/application/** +app))
+            function*
+              "__99+[Something else]_block_invoke_2"
+          frame (non app frame)
+            function*
+              "__00+[Something else]_block_invoke_2"
+--------------------------------------------------------------------------
+default:
+  hash: null
+  component:
+    default (threads of system take precedence)
+      message (threads of system take precedence)
+        "Foo"
+--------------------------------------------------------------------------
+system:
+  hash: "ff6c4ee7c54f118a9647ee86f0c2b0b0"
+  component:
+    system*
+      threads*
+        stacktrace*
+          frame* (marked in-app by stack trace rule (family:native package:**/containers/bundle/application/** +app))
```

Comment:
This is the behaviour that I want.

I tested out the change in [this other PR](https://github.com/getsentry/sentry/pull/61592/files#diff-9b9f840201a8d2ee313031ee6bdd8d43aef2a7b7f0159e56849346b9b4f29191L69) which uses the changes on this PR + removing `family:native function:_dispatch_client_callout category=threadbase` (which we need more discussion for).
<img width="785" alt="image" src="https://github.com/getsentry/sentry/assets/44410/74c06ea8-bc2c-426d-ac9d-638e5e404d45">